### PR TITLE
chore(java-buildpack): Add java to CodeQL in CI

### PR DESCRIPTION
**Pull request by menehune23**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -35,11 +35,21 @@ jobs:
...
       matrix:
-        language: [ 'go', 'python' ]
+        language:
+        - name: 'go'
+          path: 'buildpacks'
+        - name: 'python'
+          path: 'invokers/python'
+        - name: 'java'
+          path: 'invokers/java'
         # Re-add Java into main when it works
```
Comment (andrew-su) [inline .github/workflows/codeql-analysis.yml:45]: Do we want to remove this comment?

**Review by andrew-su — APPROVED** [review]
LGTM, just remove the comment about Java!

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (menehune23) [reply .github/workflows/codeql-analysis.yml:45]: > LGTM, just remove the comment about Java!

@andrew-su Thanks for the fast review. I'm actually not done with this PR and the above is just an experiment -- meant to put it in draft.

Noted about the java comment though and will remove! 👍

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -56,19 +55,15 @@ jobs:
...
-    #    and modify them (or add more) to build your code if your project
-    #    uses a compiled language
-
-    #- run: |
-    #   make bootstrap
-    #   make release
+    - if: matrix.language == 'java'
+      run: |
+        cd invokers/java
+        mvn -B -DnewVersion=$(cat ./VERSION) -DgenerateBackupPoms=false versions:set
```
Comment (menehune23) [inline .github/workflows/codeql-analysis.yml:65]: These `mvn` commands taken from the invoker dockerfile

File: invokers/java/package.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
Comment (menehune23) [inline invokers/java/package.sh:1]: We need this in multiple places. At first, I tried to just add a make target, but that needs the entire repo in order to resolve the includes. Extracting to a simple script was the next simplest thing.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (menehune23) [reply .github/workflows/codeql-analysis.yml]: Extracted out to keep this in sync with dockerfile

**Review by bbtong — APPROVED** [review]
LGTM

**Pull request merged**